### ISSUE TITLE
Upgrade flutter/flutter cipd package to Xcode 13

### ISSUE THREAD

**Issue by jmagman** [flutter/flutter#91633]
Add new cipd package for Xcode 13 with any necessary old simulators and SDKs.  Adopt in flutter/flutter.
https://github.com/flutter/flutter/blob/3257ae8815f4fcab9e53a4755f4bf76116e48be9/.ci.yaml#L70

**Comment by jmagman** [comment]
See engine builder attempts at https://github.com/flutter/flutter/issues/85555

**Comment by jmagman** [comment]
Blocked by https://github.com/flutter/flutter/issues/94078.

**Comment by jmagman** [comment]
flutter/flutter staging tests on macOS 12.0.1 and Xcode 13 seem to all be passing.
https://luci-milo.appspot.com/p/flutter/builders/staging/Mac_ios_staging%20ios_app_with_extensions_test/626

@godofredoc can you confirm?  If it all looks good I think we should upgrade the rest of the framework pool macOS version and use the Xcode 13a233 package.  

Still working on getting the engine tasks passing: https://github.com/flutter/flutter/issues/85555#issuecomment-983350700

**Comment by jmagman** [comment]
@godofredoc I'm going to assign this to you, I think the framework pools are ready to switch to macOS 12.0.1 and Xcode 13, I'm not sure what that entails other than editing the `.ci.yaml`.

**Comment by godofredoc** [comment]
This will require some planning and PSAs to work with limited capacity while we transition hosts and re-populate hashes.

**Comment by jmagman** [comment]
The prod pool is now running Xcode 13!
https://ci.chromium.org/p/flutter/builders/prod/Mac_ios%20complex_layout_scroll_perf_ios__timeline_summary/2994
`{ "sdk_version": "13a233" }`

**Comment by godofredoc** [comment]
Yes, although we have migrated only half of the capacity so far.

**Comment by jmagman** [comment]
This was completed.
